File: server/object/dto/PostServerGroupRequest.go
package dto

import (
	"promise/base"
	"promise/server/object/model"
)

// PostServerGroupRequest is the DTO for post server group.
type PostServerGroupRequest struct {
	Name        string `json:"Name"`
	Description string `json:"Description"`
}

// NewInstance creates a new instance.
func (dto *PostServerGroupRequest) NewInstance() base.RequestInterface {
	return new(PostServerGroupRequest)
}

// IsValid return if the request is valid.
func (dto *PostServerGroupRequest) IsValid() *base.ErrorResponse {
	return nil
}

// String return the name for debug.
func (dto PostServerGroupRequest) String() string {
	return dto.Name
}

// ToModel convert the DTO to model.
func (dto *PostServerGroupRequest) ToModel() base.ModelInterface {
	ret := model.ServerGroup{}
	ret.Category = base.CategoryServerGroup
	ret.Name = dto.Name
	ret.Description = dto.Description
	return &ret
}

File: integration-test/src/test/java/com/promise/integrationtest/servergroup/dto/PostServerGroupRequest.java
package com.promise.integrationtest.servergroup.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * The DTO of post server group.
 */
public class PostServerGroupRequest
{
    @JsonProperty("Name")
    private String name;
    @JsonProperty("Description")
    private String description;

    public PostServerGroupRequest() {
    	
    }
    
    public PostServerGroupRequest(String name, String description) {
    	this.name = name;
    	this.description = description;
    }
    
    public String getName()
    {
        return name;
    }

    public void setName(String name)
    {
        this.name = name;
    }

    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }
}
